test/unit/StakingPool/depositTo.js

```diff
@@ -0,0 +1,807 @@
+
+    const userBalanceBefore = await nxm.balanceOf(user.address);
+    const tokenControllerBalanceBefore = await nxm.balanceOf(tokenController.address);
+
+    const depositToParams = tranches.map(e => {
+      return {
+        amount,
+        trancheId: e,
+        tokenId,
+        destination,
+      };
+    });
+
+    await stakingPool.connect(user).depositTo(depositToParams);
+
+    const totalAmount = amount.mul(tranches.length);
+
+    const userBalanceAfter = await nxm.balanceOf(user.address);
+    const tokenControllerBalanceAfter = await nxm.balanceOf(tokenController.address);
+    expect(userBalanceAfter).to.equal(userBalanceBefore.sub(totalAmount));
+    expect(tokenControllerBalanceAfter).to.equal(tokenControllerBalanceBefore.add(totalAmount));
+
+    let totalStakeShares = BigNumber.from(0);
+    for (let depositNftId = 1; depositNftId <= tranches.length; depositNftId++) {
+      const trancheId = tranches[depositNftId - 1];
+      const deposit = await stakingPool.deposits(depositNftId, trancheId);
+
+      const newRewardShares = await getNewRewardShares({
+        stakingPool,
+        initialStakeShares: totalStakeShares,
+        stakeSharesIncrease: deposit.stakeShares,
+        initialTrancheId: trancheId,
+        newTrancheId: trancheId,
+      });
+      const newStakeShares =
+        depositNftId === 1 ? Math.sqrt(amount) : amount.mul(totalStakeShares).div(amount.mul(depositNftId - 1));
+
+      expect(deposit.pendingRewards).to.equal(0);
+      expect(deposit.lastAccNxmPerRewardShare).to.equal(0);
+      expect(deposit.stakeShares).to.equal(newStakeShares);
+      expect(deposit.rewardsShares).to.be.approximately(newRewardShares.toNumber(), 1);
+
+      const managerDeposit = await stakingPool.deposits(0, trancheId);
+      expect(managerDeposit.rewardsShares).to.equal(
+        deposit.rewardsShares.mul(initialPoolFee).div(POOL_FEE_DENOMINATOR),
+      );
+
+      const tranche = await stakingPool.tranches(trancheId);
+      expect(tranche.stakeShares).to.equal(deposit.stakeShares);
+      expect(tranche.rewardsShares).to.equal(deposit.rewardsShares.add(managerDeposit.rewardsShares));
+
+      totalStakeShares = totalStakeShares.add(deposit.stakeShares);
+    }
+  });
+
+  it('emits StakeDeposited event', async function () {
+    const { stakingPool } = this;
+    const {
+      members: [user],
+    } = this.accounts;
+
+    const { amount, tokenId, destination, depositNftId } = depositToFixture;
+
+    const { firstActiveTrancheId } = await getTranches();
+
+    await expect(
+      stakingPool.connect(user).depositTo([
+        {
+          amount,
+          trancheId: firstActiveTrancheId,
+          tokenId,
+          destination,
+        },
+      ]),
+    )
+      .to.emit(stakingPool, 'StakeDeposited')
+      .withArgs(user.address, amount, firstActiveTrancheId, depositNftId);
+  });
+
+  it.skip('reverts if provided tokenId is not valid', async function () {
```

Comment:
This test does not revert. There is a TODO in the code to validate this. Should we skip this test as it is now and addressed the fix later in a separate issue?

Reply:
you can add validation in the contract for this one, looks like a quick one

Reply:
@patitonar The change suggestion makes sense, very important to keep the comment as well otherwise it looks like a no-op. Feel free to apply the chane in this PR.

Reply:
Applied the improvement - including the comment- here https://github.com/NexusMutual/smart-contracts/pull/495/commits/79d213f2c88af227e35eeaa271a01f149baffdf3